### PR TITLE
update prettier/linter setup and workflows

### PR DESCRIPTION
**Pull request by RChandler234**
## Changes

Get Prettier and Linters working again

Some info on the linting rules Angular adds in by default: https://github.com/angular-eslint/angular-eslint/blob/main/packages/eslint-plugin-template/src/configs/README.md

I removed the react linting rules we still had in there from the og creation of the app (lol), and removed the angular accessibility linting rules from our eslintrc.json that angular adds by default (just felt like a bit much). The rest of the rules added by angulareslint recommended seemed pretty reasonable things mostly related to style

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `package-lock.json` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: .github/workflows/lint-check.yml [hunk 1]
```diff
@@ -17,14 +17,15 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
-        with:
```
Comment (bracyw) [inline .github/workflows/lint-check.yml:20]: do we just not have submodules anymore or does the linting not need to happen on them... why no more submodules?

File: angular-client/package.json [hunk 2]
```diff
@@ -31,23 +33,33 @@
     "apexcharts": "^3.44.0",
     "mapbox-gl": "^3.1.0",
     "material-design-icons": "^3.0.1",
-    "ng-apexcharts": "^1.8.0",
+    "ng-apexcharts": "1.8.0",
```
Comment (bracyw) [inline angular-client/package.json:36]: why no more updates to compatible versions?

File: angular-client/src/pages/charging-page/components/battery-info-display/battery-info-display.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ import Storage from 'src/services/storage.service';
   templateUrl: './battery-info-display.html',
   styleUrls: ['./battery-info-display.css']
 })
-export class BatteryInfoDisplay {
+export class BatteryInfoDisplayComponent implements OnInit {
```
Comment (bracyw) [inline angular-client/src/pages/charging-page/components/battery-info-display/battery-info-display.ts:11]: Why do some of these now implement OnInit (is it just convention through prettier for angular?)

**Review by bracyw — APPROVED** [review]
LGTM everything works on my end.

File: .github/workflows/lint-check.yml (hunk 1 above)
Comment (RChandler234) [reply .github/workflows/lint-check.yml:20]: Both actually, the linting is only happening on the angular client now (which never had to mess with submodules), and we removed the odyssey base submodule from scylla to simplify things

File: angular-client/src/pages/charging-page/components/battery-info-display/battery-info-display.ts (hunk 3 above)
Comment (RChandler234) [reply angular-client/src/pages/charging-page/components/battery-info-display/battery-info-display.ts:11]: Yeah (it's actually the linter tho), it just makes the typing more explicit for the Components that implement the OnInit method, no actual functionality changes or anything

File: angular-client/package.json (hunk 2 above)
Comment (RChandler234) [reply angular-client/package.json:36]: This was giving me some annoying dependency errors, removing it was the quickest fix

**Pull request merged**